src/SDKs/Azure.Core/data-plane/Azure.Core/Pipeline/HttpClientTransport.cs

```diff
@@ -57,6 +56,17 @@ internal static bool TryGetHeader(HttpHeaders headers, HttpContent content, stri
             return false;
         }
 
+        internal static bool TryGetHeader(HttpHeaders headers, HttpContent content, string name, out IEnumerable<string> values)
+        {
+            if (headers.TryGetValues(name, out values) ||
```

Comment:
You may want to consider simplifying to something like:

`return ((headers.TryGetValues(name, out values) || (content != null && content.Headers.TryGetValues(name, out values))));`